tsconfig.json

```diff
@@ -23,6 +23,5 @@
     "isolatedModules": true,
     "skipLibCheck": true
   },
-  "files": ["src/index.ts"],
-  "include": ["**/*.{js,ts}"]
```

Comment:
Not sure why, but `{a,b}` doesn't appear to work. I get

> error TS18003: No inputs were found in config file '/Users/gedge/src/github.com/mobxjs/mobx-state-tree/tsconfig.json'. Specified 'include' paths were '["**/*.{js,ts}"]' and 'exclude' paths were '["lib/"]'.

I also dropped `files`, because it seemed redundant with "include", but that changed how things get emitted, hence the changes to the build. Do we have any good tests to verify integrity of the build?

Reply:
Not that I know of, unfortunately. I think some of the tests run against the build itself, so I think that:

```sh
bun run build
bun run test:all
```

Is generally indicative of it working. But I am not very confident in this.

The output files look basically the same when I compare the file structure between `master` and this branch. I'm inclined to just merge this and pull it into the `7.0.0` release so we can use the preview builds and breaking semantics to protect against any regressions here without impacting current users who are auto-updating minor/patch versions.